services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java

```diff
@@ -148,32 +149,35 @@
-    query = "SELECT COUNT(a) FROM GamificationActionsHistory a where a.ruleId = :challengeId"
+    query = "SELECT COUNT(a) FROM GamificationActionsHistory a where a.ruleEntity.id = :challengeId"
 )
 @NamedQuery(
     name = "GamificationActionsHistory.countAnnouncementsByChallengeAndEarnerType",
-    query = "SELECT COUNT(a) FROM GamificationActionsHistory a where a.ruleId = :challengeId AND a.earnerType = :earnerType"
+    query = "SELECT COUNT(a) FROM GamificationActionsHistory a where a.ruleEntity.id = :challengeId AND a.earnerType = :earnerType"
 )
 @NamedQuery(
     name = "GamificationActionsHistory.findAllAnnouncementByChallenge",
-    query = "SELECT a FROM GamificationActionsHistory a where a.ruleId = :challengeId order by a.id desc"
+    query = "SELECT a FROM GamificationActionsHistory a where a.ruleEntity.id = :challengeId order by a.id desc"
 )
 @NamedQuery(
     name = "GamificationActionsHistory.findAllAnnouncementByChallengeByEarnerType",
-    query = "SELECT a FROM GamificationActionsHistory a where a.ruleId = :challengeId AND a.earnerType = :earnerType order by a.id desc"
+    query = "SELECT a FROM GamificationActionsHistory a where a.ruleEntity.id = :challengeId AND a.earnerType = :earnerType order by a.id desc"
 )
 @NamedQuery(
     name = "GamificationActionsHistory.findAllAnnouncementByChallengeByDate",
-    query = "SELECT a FROM GamificationActionsHistory a where a.ruleId = :challengeId AND a.createdDate >= :fromDate AND a.createdDate < :toDate order by a.id desc"
+    query = "SELECT a FROM GamificationActionsHistory a where a.ruleEntity.id = :challengeId AND a.createdDate >= :fromDate AND a.createdDate < :toDate order by a.id desc"
 )
 @NamedQuery(
     name = "GamificationActionsHistory.findAllAnnouncementByChallengeByDateByEarnerType",
-    query = "SELECT a FROM GamificationActionsHistory a where a.ruleId = :challengeId AND a.createdDate >= :fromDate AND a.createdDate < :toDate AND a.earnerType = :earnerType order by a.id desc"
+    query = "SELECT a FROM GamificationActionsHistory a where a.ruleEntity.id = :challengeId AND a.createdDate >= :fromDate AND a.createdDate < :toDate AND a.earnerType = :earnerType order by a.id desc"
 )
 @NamedQuery(
     name = "GamificationActionsHistory.findMostRealizedRuleIds",
-    query = "SELECT a.ruleId FROM GamificationActionsHistory a where a.type= :type AND (a.domainEntity.audienceId in (:spacesIds) OR a.domainEntity.audienceId = null ) AND a.createdDate >= :fromDate AND a.createdDate < :toDate " +
-            "group by a.ruleId order by count(*) DESC"
+    query = "SELECT a.ruleEntity.id FROM GamificationActionsHistory a where a.type= :type" +
+            " AND (a.domainEntity.audienceId in (:spacesIds) OR a.domainEntity.audienceId = null )" +
+            " AND a.createdDate >= :fromDate AND a.createdDate < :toDate" +
+            " AND a.ruleEntity.startDate <= :nowDate AND a.ruleEntity.endDate >= :nowDate" +
+            " group by a.ruleEntity.id order by count(*) DESC"
```

Comment:
Now, that we upgraded to JPA 2.2, we can use in JPQL `JOIN ... ON ...` for better performances and code clearness.
In addition using `= null` isn't recommended, we should use `IS NULL`.

```suggestion
    query = "SELECT r.id FROM GamificationActionsHistory a" +
        " JOIN a.ruleEntity r " +
        " ON  (r.startDate IS NULL OR r.startDate <= :nowDate)" +
        " AND (r.endDate IS NULL OR r.endDate >= :nowDate)" +
        " JOIN a.domainEntity d " +
        " ON d.audienceId IS NULL OR d.audienceId IN (:spacesIds)" +
        " WHERE a.type= :type" +
        " AND a.createdDate >= :fromDate AND a.createdDate < :toDate" +
        " group by r.id order by count(*) DESC"
```

Reply:
Done!